### ISSUE TITLE
Resource like storage account should transform name and prefixes to lower

### ISSUE THREAD

**Issue by invisibleninja06** [gsoft-inc/terraform-azurerm-naming#3]
If I do not specify `ToLower()` on each prefix and the name the regex fails, with reason, since the module is not forcing all input values ToLower. 

Would be ideal if the module did the legwork.
In this way I can pass name and prefixes that may contain capitals, acceptable for other resources, but will get lower cased when used for storage account.

**Comment by yohanb** [comment]
Yeah that would take a bit of time to implement since it would have to be specific for resources that don't allow uppercase characters.  I'll have a look but it's not a "quick win".

**Comment by haroldwongms** [comment]
Also, if a name or prefix has invalid characters like hyphens or periods in the name, this module fails with no matches. 
suggestion for storage module:
result = lower(join("", regexall("[a-z0-9]+", module.storage_account.result)))
